script/os-autoinst-openvswitch

```diff
@@ -16,229 +16,28 @@ Show this help.
-    return if (@ver != @min_ver);
-
-    return sprintf("%04d%04d%04d", @ver) >= sprintf("%04d%04d%04d", @min_ver);
-}
-
-dbus_method("set_vlan", ["string", "uint32"], ["int32", "string"]);
-sub set_vlan ($self, $tap, $vlan) {
-    my $return_output;
-    my $return_code = 1;
-    my $ovs_vsctl_error;
-    my $ovs_vsctl_output;
-    my @cmd;
-
-    ($return_code, $return_output) = _ovs_check($tap, $vlan, $self->{BRIDGE});
-
-    unless ($return_code == 0) {
-        print STDERR $return_output . "\n";
-        return ($return_code, $return_output);
-    }
-
-    @cmd = ('ovs-vsctl', 'set', 'port', $tap, "tag=$vlan");
-    if (check_min_ovs_version('2.8.1')) {
-        push(@cmd, 'vlan_mode=dot1q-tunnel');
-    }
-
-    # Connect tap device to given vlan
-    ($return_code, $ovs_vsctl_error, $ovs_vsctl_output) = _cmd(@cmd);
-
-    print STDERR $ovs_vsctl_error if length($ovs_vsctl_error) > 0;
-    print $ovs_vsctl_output if length($ovs_vsctl_output) > 0;
-    return $return_code, $ovs_vsctl_error unless $return_code == 0;
-
-    $return_code = system('ip', 'link', 'set', $tap, 'up');
-    return $return_code, $return_code != 0 ? "Failed to set $tap up " : '';
-}
-
-dbus_method("unset_vlan", ["string", "uint32"], ["int32", "string"]);
-sub unset_vlan ($self, $tap, $vlan) {
-    my $return_output;
-    my $return_code = 1;
-    my $ovs_vsctl_error;
-    my $ovs_vsctl_output;
-
-    ($return_code, $return_output) = _ovs_check($tap, $vlan, $self->{BRIDGE});
-
-    unless ($return_code == 0) {
-        print STDERR $return_output . "\n";
-        return ($return_code, $return_output);
-    }
-
-    # Remove tap device to given vlan
-    ($return_code, $ovs_vsctl_error, $ovs_vsctl_output) = _cmd('ovs-vsctl', 'remove', 'port', $tap, 'tag', $vlan);
-
-    print STDERR $ovs_vsctl_error if length($ovs_vsctl_error) > 0;
-    print $ovs_vsctl_output if length($ovs_vsctl_output) > 0;
-    return $return_code, $return_code != 0 ? $ovs_vsctl_error : '';
-}
-
-dbus_method("show", [], ["int32", "string"]);
-sub show ($self) {
-    my ($return_code, undef, $ovs_vsctl_output) = _cmd('ovs-vsctl', 'show');
-    return $return_code, $ovs_vsctl_output;
-}
-
 ################################################################################
 package main;
 use Mojo::Base -strict, -signatures;
 
 use Net::DBus;
 use Net::DBus::Reactor;
+use FindBin '$Bin';
+use lib "$FindBin::Bin/lib";
 
-my $bus = Net::DBus->system;
+sub parse_args () {
+    my %options;
+    GetOptions(\%options, 'help|h|?') or usage(1);
+    usage(0) if $options{help};
+    return \%options;
+}
 
-my $service = $bus->export_service("org.opensuse.os_autoinst.switch");
-my $object = OVS->new($service);
+sub main ($args) {
+    my $bus = Net::DBus->system;
+    my $service = $bus->export_service("org.opensuse.os_autoinst.switch");
+    my $object = OVS->new($service);
 
-Net::DBus::Reactor->main->run;
+    Net::DBus::Reactor->main->run;
+}
 
-exit 0;
+main(parse_args()) unless caller();
```

Comment:
Like mentioned before, that approach is useful if you want to test the script itself by treating it as a module, loading it directly via require.
Instead, you're moving everything into a real module anyway.

Reply:
Gotcha, changed the way I test